pepsico/projections/maproom.py

```
@@ -168,44 +165,307 @@ def pick_location(n_clicks, click_lat_lng, latitude, longitude):
         return [lat, lng], lat, lng
 
 
+    def plot_ts(ts, name, color, start_format, units):
+        return pgo.Scatter(
+            x=ts["T"].dt.strftime(STD_TIME_FORMAT),
+            y=ts.values,
+            customdata=ts["seasons_ends"].dt.strftime("%B %Y"),
+            hovertemplate=("%{x|"+start_format+"}%{customdata}: %{y:.2f}" + units),
+            name=name,
+            line=pgo.scatter.Line(color=color),
+            connectgaps=False,
+        )
+
+
     @APP.callback(
-        Output("cdf_graph", "figure"),
+        Output("local_graph", "figure"),
         Input("loc_marker", "position"),
+        Input("model", "value"),
+        Input("variable", "value"),
+        Input("region", "value"),
+        Input("submit_season","n_clicks"),
+        State("start_month", "value"),
+        State("end_month", "value"),
     )
-    def local_plots(marker_pos):
+    def local_plots(marker_pos, model, variable, region, n_clicks, start_month, end_month):
         lat = marker_pos[0]
         lng = marker_pos[1]
-        return pingrid.error_fig("REPLACE ME")
+        start_month = ac.strftimeb2int(start_month)
+        end_month = ac.strftimeb2int(end_month)
+        histo = ac.read_data("historical", model, variable, region)
+        picont = ac.read_data("picontrol", model, variable, region)
+        ssp126 = ac.read_data("ssp126", model, variable, region)
+        ssp370 = ac.read_data("ssp370", model, variable, region)
+        ssp585 = ac.read_data("ssp585", model, variable, region)
+        # Should I make this a xr.ds?
+        data_list = [histo, picont, ssp126, ssp370, ssp585]
+        try:
+            if any([var is None for var in data_list]):
+                return pingrid.error_fig(
+                    error_msg="Data missing for this model or variable"
```

Do we really need to fail if one variable is missing? Would it not be preferable to display whatever variables are available?

---

There is not supposed to be anything missing  so... it's a moot point I guess... and if one thing is missing, something is wrong... So... I don't know...

---

A verification could be implemented at the start to check that everything is in order (at the data/files level, ensuring there is data), and consequently display messages or alerts to the user if any variable or model is failing, allowing them to use the ones that are not failing. This is just an idea that can be implemented (or not) later on; I don't see it as urgent at the moment.